vmware-tanzu/tanzu-framework
# Remove packages/{management,standalone} dirs and move contents to parent dir

### What this PR does / why we need it

Remove packages/{management,standalone} directories and move their contents to the packages directory. 

We need this because Carvel packages are expected to be in the `packages` directory and not a subdirectory of `packages`.

**Other Notables**
- I did update `pkg/v1/providers/tests/unit/fixtures`. Should the `management` be removed from `imagePath:sandbox/packages/management/repo`? I kept it in for now because it's still a valid unit test and does not affect production logic.
- I kept the repo directories `standalone` and `management` in `build/package-repo-bundles`. 
- Repo bundles now have the subversion in the tar.gz and Package CR file names (previously, only had version)
- For package bundles (but not for repo bundles because the Repository struct in `hack/packages/package-tools/cmd/packageValues.go` does not have a subVersion field), the `PackageSubVersion` is taken from `package-values.yaml` The `subVersion` flag overrides `packageSubVersion`.
- Package CR and tarballs also include the `PackageSubVersion` if provided and not overridden by the `subVersion` flag

### Which issue(s) this PR fixes


Fixes #1782

### Describe testing done for PR

I manually ran on my MacBook Pro the following`make` targets and ensured that packages built and got pushed to a container registry without problems. To repeat my test, make sure to `export OCI_REGISTRY` and set it to a registry you have read and write access to. Here are some of the commands I was able to run without error:
   - make create-package PACKAGE_NAME=pr-test-pkg
   - PACKAGE_NAME=tkg make package-bundle 
   - PACKAGE_SUB_VERSION="may.03.22" PACKAGE_NAME=standalone-plugins make package-bundle
   - PACKAGE_SUB_VERSION="may.03.22" PACKAGE_NAME=capabilities make package-bundle
   - REPO_BUNDLE_SUB_VERSION="may.03.22" PACKAGE_REPOSITORY=standalone make package-repo-bundle  
   - PACKAGE_SUB_VERSION="may.03.22" PACKAGE_BUNDLES=standalone-plugins,capabilities make push-package-bundles 
   - REPO_BUNDLE_SUB_VERSION="may03.22" make push-package-repo-bundle 
   - make package-vendir-sync
        - This test resulted in a correction to the `- path: webhook-manifests.lib.yaml` in `packages/addons-manager-v2/vendir.yml#L58`. /cc @vijaykatam 
   - PACKAGE_SUB_VERSION="may.03.22" PACKAGE_REPOSITORY=standalone make package-push-bundles-repo

@tylerschultz  for pointed out this test
   - cd hack/provider-sync-tools && make validate

Here is what the new directory structure looks like in the `build` directory:
![build-directory-structure](https://user-images.githubusercontent.com/17328443/169099510-b58e7d51-1500-4bb4-b1ac-8574ec711e62.png)

### Release note

```release-note
    Removed packages/{management,standalone} directories and move their contents to the packages directory. 
```

### PR Checklist



      


### Additional information

#### Special notes for your reviewer
